src/browser/BrowserEntrySaveDialog.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+*  Copyright (C) 2013 Francois Ferrand
+*  Copyright (C) 2018 Sami Vänttinen <sami.vanttinen@protonmail.com>
+*  Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+*
+*  This program is free software: you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation, either version 3 of the License, or
+*  (at your option) any later version.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "BrowserEntrySaveDialog.h"
+#include "ui_BrowserEntrySaveDialog.h"
+
+BrowserEntrySaveDialog::BrowserEntrySaveDialog(QWidget* parent)
+    : QDialog(parent)
+    , ui(new Ui::BrowserEntrySaveDialog())
+{
+    this->setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
+
+    ui->setupUi(this);
+    connect(ui->okButton, SIGNAL(clicked()), this, SLOT(accept()));
+    connect(ui->cancelButton, SIGNAL(clicked()), this, SLOT(reject()));
+
+    ui->itemsList->setSelectionMode(QAbstractItemView::SingleSelection);
+    ui->label->setText(QString(tr("You have multiple databases open.\n"
+                                  "Please select the correct database for saving credentials.")));
+}
+
+BrowserEntrySaveDialog::~BrowserEntrySaveDialog()
+{
+}
+
+int BrowserEntrySaveDialog::setItems(QList<DatabaseWidget*>& databaseWidgets, DatabaseTabWidget* dbTabWidget) const
+{
+    uint counter = 0;
+    int activeIndex = -1;
+    for (const auto i : databaseWidgets) {
+        QListWidgetItem* item = new QListWidgetItem();
+        item->setText(dbTabWidget->getTabName(i, true));
```

Comment:
As discussed in a later comment, I would rather this come directly from the databaseWidget itself. We should not be extending the databaseTabWidget anymore.